en/ui-system/components/engine/ui-batch.md

```diff
@@ -1,19 +1,91 @@
+| Textures share the same source and sampler type | The most common issue that prevents batching is a mismatch between textures. For example, Sprites and Labels are unable to be batched due to different sampler types. It is recommended to use a general workflow to achieve optimal batching results in Cocos Creator. Please refer to the **Batching Workflow Guidelines** section below for more details. |
 
-Combined with the above description of batching conditions, we can use some methods to achieve better batching methods, but it should be noted that the rendering data collection of 2D rendering components is a node-tree-based rendering method, and some components will interrupt the batching. The three middleware components, TiledMap, Spine and DragonBones, follow their own internal batching mechanism, so the above components can't be batched and will interrupt the batching of other components, so users need to manage the node tree layout in modules to achieve better batching.
+## Batching Workflow Guidelines
+
+> **Note**: Cocos Creator renders components based on the hierarchy of the **Node Tree** structure. Batching is prone to be interrupted when encountering a node that is prohibited from batching, thus preventing the rest of the nodes in the tree to be batched. Users are encouraged to manage their Node Tree structure to achieve an optimized outcome. 
+
+Components that are prohibited from batching include:
+
+- Mask
+- Graphics
+- UIMeshRenderer
+
+These components are prohibited from batching due to utilization of a different material type and different data structures.
+
+Components below are also prohibited from render batching but follows their separate internal batching mechanism:
+
+- TiledMap
+- Spine
+- DragoneBones
+
+Spites and Labels are in general unable to be batched together due to different texture types, but this can be circumvented by combining textures. The following is a workaround solution to achieve the same goal:
+- For sprites, combine textures with [Auto Atlas](../../../asset/auto-atlas.md) and [Dynamic Atlas](../../../advanced-topics/dynamic-atlas.md). Texture Atlases can be batched with other components as long as the prerequisites are met.
+- For labels, create a bitmap cache to combine textures allowing them to be batched with sprites. However, it is ill-advised to frequently alter the content of label texts once bitmap caches are created.
+
+In summary, it is recommended to optimize the Node Tree structure in conjunction with combing textures with Auto Atlas, Dynamic Atlas and bitmap caches to achieve ideal batching results.
+
+## MeshBuffer Guidelines
+
+Please be reminded that it is required for vertex data to be transferred in the same MeshBuffer for batching to be successful. The following scenarios will result in switching between MeshBuffers:
```

Comment:
MeshBuffer is an internal data type used to store vertex and index data in 2D renderable components. Please be reminded that it is required for vertex data to be transferred in the same MeshBuffer for being batched successfully. The following scenarios will result in switching between MeshBuffers: